### ISSUE TITLE
DataTable: No width limit when condensing or expanding columns  

### ISSUE THREAD

**Issue by jocelyn-dunkley** [commercetools/ui-kit#2564]
**Describe the bug**

1. I am allowed to expand the column beyond the screen width. As a result I cannot edit the width again because the adjuster is beyond the screen size. So I must expand my own browser size and the cycle continues
2. Along with issue 1, I can expand the width of the 2nd to last column so much that the last column goes off the screen and I can't access it anymore
3. I am allowed to condense the width of the column much smaller than I expect. But there is still padding on the left side and none on the right. When I hover over the column header, more of the header title is shown but not the whole thing. 
4. If I condense multiple columns the whole table width decreases. But the DataTableManager table settings stay in the same position.  

**Expected behavior**

Issue 1: I expect to only expand the width to the screen size and nothing more. If I increase or decrease the browser width I expect the table to expand or contract along with it.

Issue 2: I expect that the last column has its limit and the 2nd to last column doesn't override it. All the columns stay within the browser width. 

Issue 3: I expect the column width limit to be when the column header text is wrapped and the sort option is still visible. 

Something like this: 
![Screenshot 2023-07-14 at 11 38 28](https://github.com/commercetools/ui-kit/assets/115018694/f5b36b02-e730-4993-9354-e4483d0beb4b)
Right now it is this:
![Screenshot 2023-07-14 at 11 40 42](https://github.com/commercetools/ui-kit/assets/115018694/f01a9f81-6d91-46f7-9a8c-bb5dab464dae)

Also the left padding should adjust with the width so the padding is always even.

Issue 4: I expect the whole table width to stick to the 100% width of the browser. The column widths will adjust within that. But the whole table width should not move.

**Screenshots**

![Kapture 2023-07-14 at 11 41 55](https://github.com/commercetools/ui-kit/assets/115018694/803108e1-5a4d-4523-a6e7-d8d0c95e3dce)
![Kapture 2023-07-14 at 11 48 39](https://github.com/commercetools/ui-kit/assets/115018694/35535a04-74df-4e6f-b31f-5fa2071bd7b2)

![Screenshot 2023-07-14 at 11 43 47](https://github.com/commercetools/ui-kit/assets/115018694/9747333c-47c8-45fa-87e0-29f18e851a65)

**Comment by jaikumar-tj** [comment]
@jocelyn-dunkley I can answer a few of the above issues. 

> **I am allowed to expand the column beyond the screen width. As a result I cannot edit the width again because the adjuster is beyond the screen size. So I must expand my own browser size and the cycle continues**

The table columns are not fixed and as a result, users can add 'n' columns according to their wishes. So even if you expand beyond the viewport, the table becomes horizontally scrollable. You don't have to expand your browser size, instead you can simply scroll the table horizontally. If we prevent that, then the table becomes too congested with so many columns.
![ezgif-1-5ab3ef7644](https://github.com/commercetools/ui-kit/assets/83634106/eb4d4582-16c9-4bfb-a6f5-ef1d09988279)

> **Along with issue 1, I can expand the width of the 2nd to last column so much that the last column goes off the screen and I can't access it anymore**

In addition to the above explanation, Some of our customers don't use the default columns, instead they add attributes and other columns via column manager, they set columns of different widths according to their usage since they can access them by scrolling horizontally.

> **Issue 3: I expect the column width limit to be when the column header text is wrapped and the sort option is still visible.**

Our Rewe customer have showed us their minimal design. They shrink the "Status column" so much, that only the dot icon was visible in our previous MC. Infact they mentioned this issue after our new design change. So they want to keep the column to be so small, that the information they need is observed without taking up more space.
<img width="61" alt="Screenshot 2023-01-17 at 11 28 48" src="https://github.com/commercetools/ui-kit/assets/83634106/ce746ae0-d072-46fd-9aa7-ee45939106d8">

**Comment by jocelyn-dunkley** [comment]
> **The table columns are not fixed and as a result, users can add 'n' columns according to their wishes. So even if you expand beyond the viewport, the table becomes horizontally scrollable. You don't have to expand your browser size, instead you can simply scroll the table horizontally. If we prevent that, then the table becomes too congested with so many columns. ![ezgif-1-5ab3ef7644](https://user-images.githubusercontent.com/83634106/253544263-eb4d4582-16c9-4bfb-a6f5-ef1d09988279.gif)**

Ah ok I see the horizontal scroller. The issue was I couldn't see it unless I scrolled down so I didn't know it was an option.  Has there been a consideration to make the table footer sticky (page change and items per page)? Because if even 1 of the rows has a lot of content the table turns into one big block of data and white space and it's hard to figure out where things end. 

 > **Our Rewe customer have showed us their minimal design. They shrink the "Status column" so much, that only the dot icon was visible in our previous MC. Infact they mentioned this issue after our new design change. So they want to keep the column to be so small, that the information they need is observed without taking up more space. <img alt="Screenshot 2023-01-17 at 11 28 48" width="61" src="https://user-images.githubusercontent.com/83634106/253542294-ce746ae0-d072-46fd-9aa7-ee45939106d8.png">**

 But what about the header title? It is still overflowing with the next header so unless you remember every column, I imagine you'll have to expand it to see what the title is then contract it again. Maybe we could add a tooltip that shows the full name on hover or at minimum add ellipses (...) so the content doesn't go into the next column. 

It brings up the questions if we should have smaller options depending on the width (like shows dots or something small instead of the stamps with text when the width is less than x). I'll chat with you privately about that.

**Comment by jaikumar-tj** [comment]
> ** as there been a consideration to make the table footer sticky (page change and items per page)?**

Yes, we used to have MAX_HEIGHT for table and the bottom pagination was always visible. But during the redesign phase, we ran into few issues and it was decided to remove the max height prop and that's why you don't see the footer.

**Comment by jocelyn-dunkley** [comment]
Could we explore the max height idea again? Because otherwise the page structure is way too fluid and you don't know when things end.
